support-workers/src/main/scala/com/gu/emailservices/PaperFieldsGenerator.scala

```diff
@@ -0,0 +1,77 @@
+package com.gu.emailservices
+
+import com.gu.i18n.Currency
+import com.gu.salesforce.Salesforce.SfContactId
+import com.gu.support.promotions.Promotion
+import com.gu.support.workers._
+import org.joda.time.LocalDate
+
+//noinspection ScalaStyle
+object PaperFieldsGenerator {
+
+  def fieldsFor(
+    subscriptionNumber: String,
+    billingPeriod: BillingPeriod,
+    user: User,
+    paymentSchedule: PaymentSchedule,
+    firstDeliveryDate: Option[LocalDate],
+    currency: Currency,
+    paymentMethod: PaymentMethod,
+    sfContactId: SfContactId,
+    directDebitMandateId: Option[String],
+    promotion: Option[Promotion]
+  ): List[(String, String)] = {
+
+    val firstPaymentDate = SubscriptionEmailFieldHelpers.firstPayment(paymentSchedule).date
+
+    val paymentFields = getPaymentFields(paymentMethod, directDebitMandateId)
+
+    val deliveryAddressFields = getAddressFields(user)
+
+    val fields = List(
+      "ZuoraSubscriberId" -> subscriptionNumber,
+      "SubscriberKey" -> user.primaryEmailAddress,
+      "EmailAddress" -> user.primaryEmailAddress,
```

Comment:
@rupertbates I've just spotted in the Membership Workflow service that this field might be called 'Address' rather than EmailAddress - https://github.com/guardian/membership-workflow/blob/a5239afe520dc3df22c27a6c9cd24169424d314d/app/model/EmailPayload.scala#L15

Reply:
That's at the next level up, here: https://github.com/guardian/support-frontend/blob/35e1d03b276188509e101993f77720a8dff732c4/support-workers/src/main/scala/com/gu/emailservices/EmailFields.scala#L27

The fields are in the payload